packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/LogObfuscator.kt

```diff
@@ -0,0 +1,248 @@
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a create API key request:
+        // `"key":"<KEY>"` becomes `"key":"***"`
+        private fun createApiKey(): LogReplacer = mapOf(
+            """(("$API_KEY_KEY"):(\s?".+?"))""".toRegex() to """"$API_KEY_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with API key request:
+        // `"key":"<KEY>"` becomes `"key":"***"`
+        private fun loginApiKey(): LogReplacer = mapOf(
+            """(("$API_KEY_KEY"):(\s?".+?"))""".toRegex() to """"$API_KEY_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with an Apple token request:
+        // `"id_token":"<TOKEN>"` becomes `"id_token":"***"`
+        private fun loginApple(): LogReplacer = mapOf(
+            """(("$ID_TOKEN_KEY"):(\s?".+?"))""".toRegex() to """"$ID_TOKEN_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with a Facebook token request:
+        // `"accessToken":"<TOKEN>"` becomes `"accessToken":"***"`
+        private fun loginFacebook(): LogReplacer = mapOf(
+            """(("$FB_ACCESS_TOKEN_KEY"):(\s?".+?"))""".toRegex() to """"$FB_ACCESS_TOKEN_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with a Google token request:
+        // `"authCode":"<TOKEN>"` becomes `"authCode":"***"`
+        // `"id_token":"<TOKEN>"` becomes `"id_token":"***"`
+        private fun loginGoogle(): LogReplacer = mapOf(
+            """(("$AUTHCODE_KEY"):(\s?".+?"))""".toRegex() to """"$AUTHCODE_KEY":"***"""",
+            """(("$ID_TOKEN_KEY"):(\s?".+?"))""".toRegex() to """"$ID_TOKEN_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with a JWT request:
+        // `"token":"<TOKEN>"` becomes `"token":"***"`
+        private fun loginJwt(): LogReplacer = mapOf(
+            """(("$TOKEN_KEY"):(\s?".+?"))""".toRegex() to """"$TOKEN_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Creates a replacer that hides parameters for custom functions
+        private fun customFunction(): LogReplacer = CustomFunctionPatternReplacer
+    }
+}
+
+// Replacer for any given feature but custom functions
+// FIXME Access and refresh tokens cannot be replaced by regex due to
+//  https://github.com/realm/realm-kotlin/issues/1284
+//  so until we figure out what is wrong, do manual string replacement and hopefully nothing will
+//  change - if the message string changes our tests will hopefully catch it
+internal class GenericRegexPatternReplacer(
+    private val patternReplacementMap: Map<Regex, String>
```

Comment:
What about taking `vararg pairs: Pair<Regex, String>` so that you could construct it directly with 
```
GenericRegexPatternReplace(
  x1 to y1,
  x2 to y2,
)
```